test/test_itensorsextensions.jl

```diff
@@ -0,0 +1,73 @@
+using ITensors:
+  ITensors,
+  ITensor,
+  Index,
+  QN,
+  dag,
+  delta,
+  inds,
+  noprime,
+  prime,
+  randomITensor,
+  replaceind,
+  replaceinds,
+  sim
+using ITensorNetworks.ITensorsExtensions: map_eigvals
+using Random
+using Test: @test, @testset
+
+Random.seed!(1234)
+@testset "ITensorsExtensions" begin
+  @testset "Test map eigvals without QNS" begin
```

Comment:
```suggestion
  @testset "Test map eigvals without QNS (eltype=$elt, dim=$n)" for elt in (Float32, Float64, Complex{Float32}, Complex{Float64}), n in [2, 3, 5, 10]
    ...
  end
```
This is usually how I write tests that loop over parameters, that way you can see what is being run in the test set log. Also good to run over single precision as well since it is easy to accidentally get that wrong and promote to double precision inside of source code.